cog/experience.py

```diff
@@ -53,25 +54,47 @@ async def on_message(self, message):
         # if message.author.id != self.bot.owner_id:
         #     return
 
-        if message.author.id in Experience._user_cooldown_:
-            # print('>> {} needs to slow down!'.format(message.author))
+        # if message.author.id in Experience._user_cooldown_:
+        #     # print('>> {} needs to slow down!'.format(message.author))
+        #     return
+
+        if (message.author.id in Experience._user_cooldown_ and not Experience._user_cooldown_[message.author.id][1]):
             return
-        
-        # potential_bonus = (_EXP_BASE * _EXP_BONUS_FACTOR - _EXP_BASE)
-        # bonus_exp = max(0, potential_bonus - potential_bonus * ())
 
-        Experience._user_cooldown_[message.author.id] = [0, Timer(self.user_cooldowned, seconds=_EXP_COOLDOWN)]
-        await Experience._user_cooldown_[message.author.id][1].start(message.author)
-        Experience._user_cooldown_[message.author.id][0] += 1
+        if (message.author.id not in Experience._user_cooldown_):
+            Experience._user_cooldown_[message.author.id] = [0, False, Timer(self.user_cooldowned, seconds=_EXP_COOLDOWN), Timer(self.user_reset_count, minutes=_EXP_BUFF_RESET)]
+            await Experience._user_cooldown_[message.author.id][3].start(message.author)
+            # the value for Experience._user_cooldown_ is [count, ableToGetXP, Timer]
+        elif (Experience._user_cooldown_[message.author.id][1]):
+            Experience._user_cooldown_[message.author.id][0] += 1
+            Experience._user_cooldown_[message.author.id][1] = False
+            Experience._user_cooldown_[message.author.id][2] = Timer(self.user_cooldowned, seconds=_EXP_COOLDOWN)
 
+        potential_bonus = (_EXP_BASE * _EXP_BONUS_FACTOR - _EXP_BASE)
         count = Experience._user_cooldown_[message.author.id][0]
-        print(count)
-        db_user_interface.modify_exp(self.bot.db_user, message.author.id, _EXP_BASE)
+        # bonus_exp = max(0, potential_bonus - potential_bonus * (count/potential_bonus)**_EXP_DECAY_FACTOR)
+        # top code is directly from Timmy's DM
+        # bottom code is a direct translation of the original formula
+        bonus_exp = max(0, _EXP_BONUS_FACTOR - (_EXP_BONUS_FACTOR - _EXP_BASE) * (count/_EXP_DECAY_UNTIL_BASE)**_EXP_DECAY_FACTOR)
+        total_exp = _EXP_BASE + bonus_exp
+
+        print(bonus_exp)
+
+        await Experience._user_cooldown_[message.author.id][2].start(message.author)
```

Comment:
I went ahead and did it myself. You no longer need to _await_ the methods for running Timer. It will also not allow you to _start()_ a timer that's already running. You must _restart()_ it.

Reply:
Wait, what do you mean I don't need to await the methods for running Timer? I don't have to put them as parameters anymore?

Reply:
As in the methods for Timer are no longer considered coroutines (no longer have async at the start of them). This means you no longer need to await them.